plugin/evm/atomic/vm/tx_gossip_test.go

```diff
@@ -0,0 +1,312 @@
+	wg.Add(1)
+
+	marshaller := atomic.TxMarshaller{}
+	onResponse = func(_ context.Context, nodeID ids.NodeID, responseBytes []byte, err error) {
+		require.NoError(err)
+
+		response := &sdk.PullGossipResponse{}
+		require.NoError(proto.Unmarshal(responseBytes, response))
+		require.Len(response.Gossip, 1)
+
+		gotTx, err := marshaller.UnmarshalGossip(response.Gossip[0])
+		require.NoError(err)
+		require.Equal(tx.ID(), gotTx.GossipID())
+
+		wg.Done()
+	}
+	require.NoError(client.AppRequest(ctx, set.Of(vm.Ctx.NodeID), requestBytes, onResponse))
+	require.NoError(vm.AppRequest(ctx, requestingNodeID, 3, time.Time{}, <-peerSender.SentAppRequest))
+	require.NoError(network.AppResponse(ctx, snowCtx.NodeID, 3, <-responseSender.SentAppResponse))
+	wg.Wait()
+}
+
+// Tests that a tx is gossiped when it is issued
+func TestAtomicTxPushGossipOutbound(t *testing.T) {
+	require := require.New(t)
+	ctx := context.Background()
+	snowCtx := snowtest.Context(t, snowtest.CChainID)
+	snowCtx.AVAXAssetID = ids.GenerateTestID()
+	validatorState := utils.NewTestValidatorState()
+	snowCtx.ValidatorState = validatorState
+	memory := avalancheatomic.NewMemory(memdb.New())
+	snowCtx.SharedMemory = memory.NewSharedMemory(snowCtx.ChainID)
+
+	pk, err := secp256k1.NewPrivateKey()
+	require.NoError(err)
+	address := pk.EthAddress()
+	genesis := vmtest.NewPrefundedGenesis(100_000_000_000_000_000, address)
+	genesisBytes, err := genesis.MarshalJSON()
+	require.NoError(err)
+
+	sender := &enginetest.SenderStub{
+		SentAppGossip: make(chan []byte, 1),
+	}
+	vm := newAtomicTestVM()
+	vm.AtomicTxPullGossiper = gossip.NoOpGossiper{}
+
+	require.NoError(vm.Initialize(
+		ctx,
+		snowCtx,
+		memdb.New(),
+		genesisBytes,
+		nil,
+		nil,
+		nil,
+		sender,
+	))
+	require.NoError(vm.SetState(ctx, snow.NormalOp))
+
+	defer func() {
+		require.NoError(vm.Shutdown(ctx))
+	}()
+
+	// Issue a tx to the VM
+	utxo, err := addUTXO(
+		memory,
+		snowCtx,
+		ids.GenerateTestID(),
+		0,
+		snowCtx.AVAXAssetID,
+		100_000_000_000,
+		pk.Address(),
+	)
+	require.NoError(err)
+	tx, err := atomic.NewImportTx(vm.Ctx, vm.CurrentRules(), vm.clock.Unix(), vm.Ctx.XChainID, address, vmtest.InitialBaseFee, secp256k1fx.NewKeychain(pk), []*avax.UTXO{utxo})
+	require.NoError(err)
+	require.NoError(vm.AtomicMempool.AddLocalTx(tx))
+	vm.AtomicTxPushGossiper.Add(tx)
+
+	gossipedBytes := <-sender.SentAppGossip
+	require.Equal(byte(p2p.AtomicTxGossipHandlerID), gossipedBytes[0])
+
+	outboundGossipMsg := &sdk.PushGossip{}
+	require.NoError(proto.Unmarshal(gossipedBytes[1:], outboundGossipMsg))
+	require.Len(outboundGossipMsg.Gossip, 1)
+
+	marshaller := atomic.TxMarshaller{}
+	gossipedTx, err := marshaller.UnmarshalGossip(outboundGossipMsg.Gossip[0])
+	require.NoError(err)
+	require.Equal(tx.ID(), gossipedTx.ID())
+}
+
+// Tests that a tx is gossiped when it is issued
+func TestAtomicTxPushGossipInbound(t *testing.T) {
+	require := require.New(t)
+	ctx := context.Background()
```

Comment:
use `testContext` here:
```suggestion
ctx, cancel := testContext(t)
defer cancel()
```